### PR TITLE
feat: show degen or ens name if available on bounty info page

### PR DESCRIPTION
**Pull request by defichopper**
See https://github.com/picsoritdidnthappen/poidh-app/issues/88

This is the updated BountyInfo page

image

### CODE REVIEW

File: src/app/context/web3.tsx [hunk 1]
```diff
@@ -575,3 +582,23 @@ export const getURI: GetURIFunction = async (claimId) => {
...
+    return null;
+  }
+
+  const degenNameContract = await getDegenNameContract();
+  const degenName = await degenNameContract.defaultNames(addr);
+  if (degenName) {
+    return `${degenName}.degen`;
+  }
+
+  return publicClient.getEnsName({ address: addr });
```
[the remark this answers is not in the document]
Comment (tempe-techie) [reply src/app/context/web3.tsx:603]: Maybe `await` would also be needed, not sure

[the remark this answers is not in the document]
Comment (tempe-techie) [reply src/app/context/web3.tsx:603]: Nevermind, `getEnsName` is a viem function, so it should work.

Comment (defichopper) [inline src/app/context/web3.tsx:603]: These two methods do the same thing - returning the primary ENS name of the address

So `lookupAddress` doesn't work either

**Review by Rhovian — APPROVED** [review]

**Comment by picsoritdidnthappen** [comment]
appreciate the help @defichopper!

**Comment by defichopper** [comment]
> appreciate the help @defichopper!

pleasure

**Comment by tempe-techie** [comment]
@defichopper Can you make a claim here? https://degen.poidh.xyz/bounty/205 

So that you can then get the bounty 🙂

**Comment by defichopper** [comment]
> @defichopper Can you make a claim here? https://degen.poidh.xyz/bounty/205
> 
> So that you can then get the bounty 🙂

Sure, I will do that now

**Comment by defichopper** [comment]
> @defichopper Can you make a claim here? https://degen.poidh.xyz/bounty/205
> 
> So that you can then get the bounty 🙂

Just created a claim here

image

**Pull request merged**